assets/common/src/batch_deploy.py

```diff
@@ -0,0 +1,320 @@
+from pathlib import Path
+
+from utils.config import AppName
+from utils.common_utils import get_mlclient
+from utils.logging_utils import custom_dimensions, get_logger
+from utils.exceptions import (
+    swallow_all_exceptions,
+    BatchEndpointInvocationError,
+    EndpointCreationError,
+    DeploymentCreationError,
+    ComputeCreationError,
+)
+
+
+MAX_INSTANCE_COUNT = 20
+
+logger = get_logger(__name__)
+custom_dimensions.app_name = AppName.DEPLOY_MODEL
+
+
+def parse_args():
+    """Return arguments."""
+    parser = argparse.ArgumentParser()
+
+    # Defaults for batch endpoint has been picked mostly from:
+    # https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-deployment-batch
+    # Some of the defaults have been tweaked to cater to large models.
+
+    # add arguments
+    parser.add_argument(
+        "--registration_details",
+        type=Path,
+        help="Json file that contains the ID of registered model to be deployed",
+    )
+    parser.add_argument(
+        "--model_id",
+        type=str,
+        help="Registered mlflow model id",
+    )
+    parser.add_argument(
+        "--inference_payload",
+        type=Path,
+        help="Folder with inference endpoint data.",
+    )
+    parser.add_argument(
+        "--endpoint_name",
+        type=str,
+        help="Name of the endpoint",
+    )
+    parser.add_argument("--deployment_name", type=str, help="Name of the the deployment")
+    parser.add_argument(
+        "--compute_name",
+        type=str,
+        help="Name of the compute cluster",
+    )
+    parser.add_argument(
+        "--size",
+        type=str,
+        help="Size of the compute cluster",
+        default="Standard_NC24s_v3",
+    )
+    parser.add_argument(
+        "--min_instances",
+        type=int,
+        default=0,
+        help="Minimum instances of the compute cluster",
+    )
+    parser.add_argument(
+        "--max_instances",
+        type=int,
+        default=3,
+        help="Maximum instances of the compute cluster",
+    )
+    parser.add_argument(
+        "--idle_time_before_scale_down",
+        type=int,
+        default=120,  # 2min
+        help="Idle time before scale down of the compute cluster",
+    )
+    parser.add_argument(
+        "--output_file_name",
+        type=str,
+        default="predictions.csv",
+        help="Name of the output file",
+    )
+    parser.add_argument(
+        "--max_concurrency_per_instance",
+        type=int,
+        default=1,
+        help="Maximum number of parallelism per instance",
+    )
+    parser.add_argument(
+        "--error_threshold",
+        type=int,
+        default=-1,
+        help="Error threshold before batch inference is aborted",
+    )
+    parser.add_argument(
+        "--max_retries",
+        type=int,
+        default=3,
+        help="Number of retries in failure",
+    )
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        default=30,  # 30sec
```

Comment:
30 is way little for most OSS models. Could we have 500 or more?

Reply:
Sure, I will change to 500, I used the documentations for the default values.